graphics/managed_surface.cpp

```diff
@@ -836,16 +840,22 @@ void ManagedSurface::clear(uint32 color) {
 		fillRect(getBounds(), color);
 }
 
+void ManagedSurface::clearPalette() {
+	if (_palette) {
+		delete _palette;
+		_palette = nullptr;
+	}
+}
+
 void ManagedSurface::grabPalette(byte *colors, uint start, uint num) const {
 	assert(start < 256 && (start + num) <= 256);
-	memcpy(colors, _palette + 3 * start, 3 * num);
+	memcpy(colors, _palette->data + 3 * start, 3 * num);
 }
 
 void ManagedSurface::setPalette(const byte *colors, uint start, uint num) {
 	assert(start < 256 && (start + num) <= 256);
-	memcpy(_palette + 3 * start, colors, 3 * num);
-
-	_paletteSet = true;
+	_palette = new Palette(start + num);
```

Comment:
This is incorrect. Usage of setPalette sometimes only sets one entry at a time.
The palette should only be allocated if nullptr and probably with 256 by default